vm/src/error.rs

```diff
@@ -49,3 +51,33 @@ impl From<EncodingError> for ActorError {
         }
     }
 }
+
+impl From<AmtError> for ActorError {
+    fn from(e: AmtError) -> Self {
+        Self {
+            fatal: false,
+            exit_code: ExitCode::ErrSerialization,
+            msg: e.to_string(),
+        }
+    }
+}
+
+impl From<HamtError> for ActorError {
+    fn from(e: HamtError) -> Self {
+        Self {
+            fatal: false,
+            exit_code: ExitCode::ErrSerialization,
+            msg: e.to_string(),
+        }
+    }
+}
+
+impl From<String> for ActorError {
+    fn from(e: String) -> Self {
+        Self {
+            fatal: false,
+            exit_code: ExitCode::ErrSerialization,
+            msg: e,
+        }
+    }
+}
```

Comment:
Probably not good to just forward all errors from this. Message receipts are very specific, and exit codes are specifically used in consensus, so forwarding all ExitCodes to a serialized error is probably incorrect. I haven't been as diligent in the other impls to make sure until the spec is finalized, but it should definitely be attempted to match error codes 1:1 for each specific case

Reply:
Yeah, that makes sense. I'll remove and just map the err where needed.